udata_front/theme/gouvfr/assets/js/components/Report/ReportModal.vue

```diff
@@ -0,0 +1,177 @@
+                    :placeholder="t('Reason of your report.\nDon\'t include any personal data.')"
+                  />
+                </div>
+                <div class="fr-modal__footer">
+                  <div class="fr-btns-group fr-btns-group--right fr-btns-group--inline-lg fr-btns-group--icon-left">
+                    <button
+                      class="fr-btn"
+                      :aria-controls="id"
+                      target="_self"
+                      @click.prevent.stop.capture="send"
+                    >
+                      {{ t("Send") }}
+                    </button>
+                  </div>
+                </div>
+              </template>
+            </div>
+          </div>
+        </div>
+      </div>
+    </dialog>
+  </Teleport>
+</template>
+<script lang="ts">
+import { getReportReasons, type ReportReason } from '../../api/reports';
+import { terms_url, title } from '../../config';
+import { required } from '../../i18n';
+import useFunctionalState from '../../composables/form/useFunctionalState';
+import SelectGroup, { type Option } from '../Form/SelectGroup/SelectGroup.vue';
+import InputGroup from '../Form/InputGroup/InputGroup.vue';
+
+export type ReportModalProps = {
+  id: string,
+};
+
+export type ReportModalForm = {
+  reason: ReportReason,
+  message: string,
+};
+</script>
+<script setup lang="ts">
+import { computed, onMounted, reactive, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { Well, getRandomId } from '@etalab/data.gouv.fr-components';
+
+defineProps<ReportModalProps>();
+
+const emit = defineEmits<{
+  (event: 'send', form: ReportModalForm): void,
+}>();
+
+const { t } = useI18n();
+const reportModalTitleId = getRandomId("modal-title");
+const modalRef = ref<HTMLDialogElement | null>();
+const showConfirm = ref(false);
+
+const initialState = {
+  reason: "",
+  message: "",
+};
+
+const form = reactive<ReportModalForm>({...initialState});
+
+const reasons = ref<Array<Option>>([]);
+
+const requiredRules = {
+  reason: { required },
+};
+
+const { getErrorText, getFunctionalState, hasError, reset, validateRequiredRules, v$ } = useFunctionalState(form, requiredRules, requiredRules);
+
+const fieldHasError = (field: string) => hasError(state, field);
+
+const state = computed(() => {
+  return {
+    reason: getFunctionalState(v$.value.reason.$dirty, v$.value.reason.$error, false),
+  };
+});
+
+function close() {
+  globalThis.dsfr(modalRef.value).modal.conceal();
+  form.message = initialState.message;
+  form.reason = initialState.reason;
+  reset();
+};
+
+function send() {
+  validateRequiredRules().then(valid => {
+    if(valid) {
+      emit('send', {...form});
+      showConfirm.value = true;
+    }
+  });
+};
+
+onMounted(async () => {
+  const reportReasons = await getReportReasons();
+  reasons.value = Object.entries(reportReasons).map<Option>(([value, label]) => {
+    return {
+      value,
+      label,
+    };
+  })
+});
```

Comment:
I think we can change the API to return this format if you prefer it :-)

Reply:
Done in https://github.com/opendatateam/udata/pull/3078